homeassistant/components/flipr/__init__.py

```diff
@@ -28,6 +53,22 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
 
     if unload_ok:
-        hass.data[DOMAIN].pop(entry.entry_id)
+        hass.data[DOMAIN].pop(CONF_ENTRY_FLIPR_COORDINATORS)
 
     return unload_ok
+
+
+def detect_invalid_old_configuration(hass: HomeAssistant, entry: ConfigEntry):
+    """Detect invalid old configuration and raise error if found."""
+
+    def find_duplicate_entries(entries):
+        values = [e.data["email"] for e in entries]
+        _LOGGER.debug("Detecting duplicates in values : %s", values)
+        return any(count > 1 for count in Counter(values).values())
```

Comment:
We are only checking if the user has more than 1 account setup, not if the email is the same

Reply:
If the user has more than one account setup with different errors, it should not be considered as an error since the user has in such a case differents fliprs corresponding to different accounts.
The problem is only if the user has 2 or more fliprs in the same account (same email) : in such a case, with the new implementation, on flipr device can be in many configurations so it generates errors at initialisation.

Reply:
What does counter do?

Reply:
In counts for each email adress, the number of occurrence.
Do you think there is an error in this code ?

Reply:
Ooooh, I didn't know the class